packages/core/src/App/Containers/Layout/header/traders-hub-onboarding.tsx

```diff
@@ -14,6 +17,9 @@ const TradersHubOnboarding = observer(() => {
     const onClickHandler = has_wallet
         ? () => {
               setWalletsOnboarding('started');
+              if (history.location.pathname !== routes.wallets) {
+                  history.push(routes.wallets);
+              }
```

Comment:
Hey @rostislav-deriv looks good on my side however there is a bug where the onboarding is being triggered when a modal is open. Here is the bug https://app.clickup.com/t/20696747/WALL-3790. Maybe what you can do is to add another check for modal in the url if the query parameter for `?modal-open=true` then maybe close modal and proceed with the onboarding.

Reply:
Hey @aizad-deriv, will add this in this PR

Reply:
Done

Reply:
> Maybe what you can do is to add another check for modal in the url if the query parameter for ?modal-open=true then maybe close modal and proceed with the onboarding.

i've tried this but doing this in core doesn't work, instead the `hide` function in wallets' `useModal` should be called since it contains some other calls beside deleting `modal-open=true` from the URL. But this can't be done in core bc this is wallets function, so i just call it pre wallets tour guide instead.